### ISSUE TITLE
internal error in syntactically incorrect "super" constructor call

### ISSUE THREAD

**Issue by TGlas** [TGlas/tscript#132]
### Description
The syntax checks of a constructor calling the super class seem to be fragile, allowing an internal error to be exposed to the user.

### To Reproduce
Die (buggy) constructor definition in the sub-class of the following example code raises an internal error:
`error in line undefined: internal parser error; TScript internal assertion failed; parse_call internal error`

```
class Fruit
{
private:
	var m_price;
	var m_weight;

public:
	constructor(price, weight)
	{
		m_price = price;
		m_weight = weight;
	}

	function price()
	{ return m_price; }

	function weight()
	{ return m_weight; }
}

class Apple:Fruit
{
private:
	var m_color;
public:
	constructor(price, weight, color):super Fruit     # ERROR IS HERE
	{
		m_color = color;
	}

	function color()
	{ return m_color;}
}
```

Steps to reproduce the behavior and what was expected to happen instead:

1. Copy the above code into the IDE
2. Try to run the code. The error occurs in the parsing phase.

What should happen instead: The parser should spit out a meaningful error message.

**Comment by TGlas** [comment]
Fixed in e219908bf88ebf8c3a16fa8f5c5a538739cb217f

**Comment by TGlas** [comment]
Fixed.
